config/runtimes/vllm-multinode-template.yaml

```diff
@@ -0,0 +1,259 @@
+  labels:
+    opendatahub.io/dashboard: "true"
+    opendatahub.io/ootb: "true"
+  annotations:
+    description: vLLM is a high-throughput and memory-efficient inference and serving engine for LLMs
+    openshift.io/display-name: vLLM ServingRuntime for KServe
+    openshift.io/provider-display-name: Red Hat, Inc.
+    tags: rhods,rhoai,kserve,servingruntime
+    template.openshift.io/documentation-url: https://github.com/opendatahub-io/vllm
+    template.openshift.io/long-description: This template defines resources needed to deploy vLLM ServingRuntime with KServe in Red Hat OpenShift AI
+    opendatahub.io/modelServingSupport: '["single"]'
+    opendatahub.io/apiProtocol: "REST"
+  name: vllm-multinode-runtime-template
+objects:
+  - apiVersion: serving.kserve.io/v1alpha1
+    kind: ServingRuntime
+    metadata:
+      name: vllm-multinode-runtime
+      annotations:
+        openshift.io/display-name: vLLM ServingRuntime for KServe
+        opendatahub.io/recommended-accelerators: '["nvidia.com/gpu"]'
+      labels:
+        opendatahub.io/dashboard: "true"
+    spec:
+      annotations:
+        prometheus.io/port: "8080"
+        prometheus.io/path: "/metrics"
+      multiModel: false
+      supportedModelFormats:
+        - autoSelect: true
+          name: vLLM
+          priority: 2
+      containers:
+        - name: kserve-container
+          image: $(vllm-image)
+          command: ["bash", "-c"]
+          args:
+            - |
+              ray start --head --disable-usage-stats --include-dashboard false 
+              # wait for other node to join
+              until [[ $(ray status | grep -c node_) -eq ${PIPELINE_PARALLEL_SIZE} ]]; do
+                echo "Waiting..."
+                sleep 1
+              done
+              ray status
+
+              export SERVED_MODEL_NAME=${MODEL_NAME}
+              export MODEL_NAME=${MODEL_DIR} 
+            
+              exec python3 -m vllm.entrypoints.openai.api_server --model=${MODEL_NAME} --served-model-name=${SERVED_MODEL_NAME}
+          env:
+            - name: RAY_PORT
+              value: "6379"
+            - name: RAY_ADDRESS
+              value: 127.0.0.1:6379
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: VLLM_CONFIG_ROOT
+              value: /tmp           
+            - name: DISTRIBUTED_EXECUTOR_BACKEND
+              value: "ray"
+            - name: DISABLE_CUSTOM_ALL_REDUCE
+              value: "true"
+            - name: MAX_SEQUENCE_LENGTH
+              value: "8192"
+            - name: MAX_NEW_TOKENS
+              value: "2048"
+            - name: MAX_BATCH_SIZE
+              value: "256"
+            - name: MAX_CONCURRENT_REQUESTS
+              value: "320"
+            - name: PORT
+              value: "8080"
+            - name: MAX_LOG_LEN
+              value: "100"
+            - name: HF_HUB_CACHE
+              value: /tmp
```

Comment:
We should remove all of these env vars, and only set `HF_HOME` instead (see existing vllm template for comparison).

`DISTRIBUTED_EXECUTOR_BACKEND` should be set as arg and port should be added aswell :

```
python3 -m vllm.entrypoints.openai.api_server <...> --port 8080 --distributed-executor-backend ray

```

Reply:
I hit an issue when I don't use `VLLM_CONFIG_ROOT` with old vllm image (quay.io/opendatahub/vllm@sha256:7f19dde68eb47abeea155f0d68d4e708f4d93cc91fc632b7a5a0de181d8d193b).  I don't have the error message now but I will leave for now. when kserve is ready to test, I am going to double check this parameter again. If error still happens, I will keep. The rest of parameters, I will remove it.

Reply:
Ah, looking at the code, it looks like it might be enough to set `VLLM_NO_USAGE_STATS=1`